public/app/features/scenes/core/types.ts

```diff
@@ -41,18 +41,24 @@ export interface SceneDataState extends SceneObjectStatePlain {
   data?: PanelData;
 }
 
-export interface SceneObject<TState extends SceneObjectState = SceneObjectState> extends Subscribable<TState> {
+export interface SceneObject<TState extends SceneObjectState = SceneObjectState> {
   /** The current state */
-  state: TState;
+  readonly state: TState;
 
   /** True when there is a React component mounted for this Object */
-  isActive?: boolean;
+  readonly isActive: boolean;
 
   /** SceneObject parent */
-  parent?: SceneObject;
+  readonly parent?: SceneObject;
 
-  /** Currently only used from root to broadcast events */
-  events: EventBus;
+  /** Subscribe to state changes */
+  subscribeToState(observer?: Partial<Observer<TState>>): Subscription;
+
+  /** Subscribe to a scene event */
+  subscribeToEvent<T extends BusEvent>(typeFilter: BusEventType<T>, handler: BusEventHandler<T>): Unsubscribable;
+
+  /** Publish an event and optionally bubble it up the scene */
+  publishEvent(event: BusEvent, bubble?: boolean): void;
```

Comment:
an alternative to this would be. 

```
getStateStream(): Observable<TState>
getEventStream<TEvent extends BusEvent>(typeFilter: BusEventType<TEvent>): Observable<TEvent>
```

I am torn. I like the readability of subscribeToState and subscribeToEvent. 

For example:

```ts
  model.getEventStream(MyCoolSceneEvent).subscribe({next: this.onMyCoolEvent});
vs
model.subscribeToEvent(MyCoolSceneEvent, this.onMyCoolEvent);
````

but pretty small difference, returning Observable directly does mean more interoperability with rxjs operators and util functions (without having to create a wrapper Observable).